controllers/pipelines/provider_controller_test.go

```
@@ -0,0 +1,101 @@
+//go:build unit
+
+package pipelines
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	config "github.com/sky-uk/kfp-operator/apis/config/v1alpha6"
+	pipelinesv1 "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha6"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Context("Provider Controller", func() {
+	var _ = Describe("constructDeployment", func() {
+		Specify("Should construct a Deployment with correct values given config and a Provider resource", func() {
```

what about incorrect values? e.g. if `ServiceContainerName` did not match the correct container in `PodTemplateSpec`?

---

Hmmm not sure... should we throw an error in that case?

---

So... try to find a container with that name, if none is found return an error, otherwise mutate it to add the env vars etc.

---

yeah I'd say we should throw an error as it would be a user misconfiguring the config and therefore they should be notified of such error at the earliest opportunity :)

---

Done in 7c1ad74

---

FYI it would error anyway, because `constructDeployment` adds the `image` on the container, and if the `image` isn't set the Deployment is invalid.

But this is better because users will know why the image isn't set.

---

FYI it would error anyway, because `constructDeployment` adds the `image` on the container, and if the `image` isn't set the Deployment is invalid.

But this is better because users will know why the image isn't set.